crates/sui-json-rpc/src/read_api.rs

```diff
@@ -125,11 +125,20 @@ impl ReadApiServer for ReadApi {
                 SuiObjectResponseError::NotExists { object_id: id },
             )),
             ObjectRead::Exists(object_ref, o, layout) => {
-                let display_fields = if options.show_display {
-                    get_display_fields(self, &o, &layout).await?
-                } else {
-                    None
-                };
+                let mut display_fields = None;
+                if options.show_display {
+                    match get_display_fields(self, &o, &layout).await {
+                        Ok(rendered_fields) => display_fields = rendered_fields,
+                        Err(e) => {
+                            return Ok(SuiObjectResponse::new(
```

Comment:
I decided to just return the object metadata, and the error string from the field that could be wrong. @damirka

Reply:
I'll add a cherry-pick for `get_display_fields` best effort in next PR.